autopush/endpoint.py

```diff
@@ -95,148 +182,277 @@
         the client will not get the updated message until reconnect. This
         should be considered a rare edge-case.
 
+Reply:
+^^^^^^
+
+.. code-block:: json
+
+    {}
+
+Errors:
+^^^^^^^
+
     :statuscode 404: `message_id` is not found.
     :statuscode 200: Message has been updated.
 
-.. http:get:: /register/(uuid:uaid)
 
-    Returns registered router data for the UAID.
+Push Service Bridge HTTP Interface
+==================================
 
-    **Example Request**
+Push allows for remote devices to perform some functions using an HTTP
+interface. This is mostly used by devices that are bridging via an
+external protocol like
+`GCM <https://developers.google.com/cloud-messaging/>`__ or
+`APNs <https://developer.apple.com/library/ios/documentation/NetworkingIntern\
+et/Conceptual/RemoteNotificationsPG/Introduction.html#//apple_ref/doc/uid/TP4\
+0008196-CH1-SW1>`__. All message bodies must be UTF-8 encoded.
 
-    .. code-block:: http
+Lexicon
+-------
 
-        GET /register/5bbc4aae-a575-4f6a-a4b3-6f84a4d06a63
-        Host: endpoint.push.com
-        Authorization: Bearer ...
+For the following call definitions:
 
-    **Example Response**
+   :{type}: The bridge type.
 
-    .. code-block:: http
+Allowed bridges are `gcm` (Google Cloud Messaging) and `apns` (Apple
+Push Notification system)
 
-        HTTP/1.1 200 OK
-        Content-Type: application/json
-        Content-Length: nnn
+   :{token}: The bridge specific public exchange token
 
-        {
-            "type": "apns",
-            "data": {
-                "token": "APNS_TOKEN_DATA",
-                ...
-            }
-        }
+Each protocol requires a unique token that addresses the remote application.
+For GCM, this is the `SenderID` and is pre-negotiated outside of the push
+service.
+
+   :{instanceid}: The bridge specific private identifier token
+
+Each protocol requires a unique token that addresses the instance of the
```

Comment:
Copy: I'm not sure what "instance of the application" means. Is it the app for that sender ID?